File: packages/igloo-ts/src/types/padding.ts
import type { Padding as WitPadding } from 'iced:app/padding@0.1.0';

export type { Padding as WitPadding } from 'iced:app/padding@0.1.0';

/**
 * Helper functions for creating Padding values.
 */
export const Padding = {
  /** Create padding with the same value on all sides */
  all(value: number): WitPadding {
    return { top: value, right: value, bottom: value, left: value };
  },

  /** Create padding with separate horizontal and vertical values */
  xy(x: number, y: number): WitPadding {
    return { top: y, right: x, bottom: y, left: x };
  },

  /** Create padding with separate values for each side */
  each(top: number, right: number, bottom: number, left: number): WitPadding {
    return { top, right, bottom, left };
  },

  /** No padding */
  none(): WitPadding {
    return { top: 0, right: 0, bottom: 0, left: 0 };
  },
} as const;

File: packages/igloo-py/src/igloo_py/types/padding.py
"""Padding type helpers."""

from wit_world.imports.padding import Padding as WitPadding


class _Padding:
    """Helper functions for creating Padding values."""

    @staticmethod
    def all(value: float) -> WitPadding:
        """Create padding with the same value on all sides."""
        return WitPadding(left=value, right=value, top=value, bottom=value)

    @staticmethod
    def xy(x: float, y: float) -> WitPadding:
        """Create padding with separate horizontal and vertical values."""
        return WitPadding(left=x, right=x, top=y, bottom=y)

    @staticmethod
    def each(top: float, right: float, bottom: float, left: float) -> WitPadding:
        """Create padding with separate values for each side."""
        return WitPadding(left=left, right=right, top=top, bottom=bottom)

    @staticmethod
    def none() -> WitPadding:
        """No padding."""
        return WitPadding(left=0, right=0, top=0, bottom=0)


# Singleton instance for convenient access
Padding = _Padding()
